feat: add scraper functionality and store with TinyDB

Kingteena opened the pull request:
### Description
Implements the scraper functionality in scraper.py and stores the data as a TinyDB database.

### Changes Made
- Fixes bug in data_fetcher.py
- Implements more functions in data_parser.py
- Iterates through all API endpoints in scraper.py and stores it in a Tinydb database
- Adds rudimentary rate limiter

### Related Issues
Implements most functionality outlined in issue #2

rayokamoto commented on src/data_parser.py:
```diff
@@ -1,20 +1,47 @@
 import data_fetcher
+import json
```
Please group and order imports by standard libs, third party libs, and local libs, per [PEP 8 guidelines](https://peps.python.org/pep-0008/#imports)

Kingteena replied:
Noted, thank you for the link to the documentation.

rayokamoto commented on src/data_parser.py:
```diff
@@ -1,20 +1,47 @@
+    data = courses.get()["data"]
+    
+    course_ids = []
+    for course in data:
```
If the `get()` function returned an empty dict (in the case of an API request error) then the "data" field will be `None`. Trying to iterate over `None` will result in errors so you'll need to do some checking to avoid bugs

Kingteena replied:
I did not implement any exception handling for when the API call fails, because I feel like the most effective thing to do is to retry the request after waiting a few seconds, which should be implemented when the rate limiter gets implemented.

rayokamoto replied:
Yeah that should be fine. I was mainly referring to checking if `data` was `None`, which it seems like you've implemented with

```py
if data is None:
    return {"courses": []}
```

rayokamoto commented on src/data_parser.py:
```diff
@@ -1,20 +1,47 @@
 
 
-# TODO: add other functions for parsing data
+# TODO: Merge get_course_details and get_course_class_list into one function in the format specified in issue #3 in the GitHub
```
```suggestion
# TODO: Merge get_course_details and get_course_class_list into one function in the format specified in issue #3 in the GitHub

```

rayokamoto commented on src/data_parser.py:
Can you add function comments using [docstrings](https://peps.python.org/pep-0008/#documentation-strings)?

rayokamoto commented on src/scraper.py:
```diff
@@ -1,5 +1,35 @@
+import data_parser
+from datetime import datetime
+from tinydb import TinyDB
+import json
```
Same comment as before on import order

rayokamoto commented:
There are some issues regarding rate limiting but we can open a new issue regarding this and it can be fixed in another PR

Kingteena commented:
I've added a commit adding a rudimentary rate limiter that can be fixed with another issue and pull request, and implemented most changes requested by @rayokamoto. 

I did not implement any exception handling for when the API call fails, because I feel like the most effective thing to do is to retry the request after waiting a few seconds, which should be implemented when the rate limiter gets implemented.
